scripts/dialog_xml2json.py

```diff
@@ -724,39 +747,66 @@ def convertAll(upperNodeJson, nodeXml):
         nodeXml (Element): Parsed XML representation to be translated
     """
     key = nodeXml.tag #key is index/selector to upperNodeJson, it is either name (e.g. generic)
+    if VERBOSE: printf("DEBUG: tag '%s'\n", nodeXml.tag)
     if type(upperNodeJson) is list:  # or an index of the last element of the array
         key = len(upperNodeJson) - 1
+    if VERBOSE: printf("DEBUG: key '%s'\n", str(key))
+    if nodeXml.get(XSI+'nil') is not None:
+        if nodeXml.get(XSI+'nil') in ["True", "true"]:
+            if VERBOSE: printf("DEBUG: Tag is None\n")
+            upperNodeJson[key] = None
+            return
+        elif nodeXml.text in ["False", "false"]:
+            pass
+        else:
+            eprintf("ERROR: Unable to parse boolean " + nodeXml.get(XSI+'nil') + "\n")
 
     if not list(nodeXml): # it has no children (subtags) - it is a terminal
+        if VERBOSE: printf("DEBUG: Tag is terminal\n")
+        if VERBOSE: printf("DEBUG:  structure '%s'\n", str(nodeXml.get('structure')))
         if nodeXml.get('structure') is not None:
             if nodeXml.get('structure') == 'emptyList':
+#                if VERBOSE: printf("DEBUG: Tag is emptyList\n")
                 upperNodeJson[key] = []
+                return
             elif nodeXml.get('structure') == 'emptyDict':
+#                if VERBOSE: printf("DEBUG: Tag is emptyDict\n")
                 upperNodeJson[key] = {}
-        elif nodeXml.text is None:
-            upperNodeJson[key] = None
-        elif nodeXml.text:  # if a single element with text - terminal (string, number or none)
+                return
+#            elif nodeXml.get('structure') == 'listItem' and nodeXml.text:
```

Comment:
create and issue from this piece of commented code, and please remove it from here

Reply:
#224